### PR TITLE
various small fixes related to open/due time changes

### PR DESCRIPTION
**Pull request by pandafulmanda**
- [x] after task save, url for calendar date should be date only
- [x] time input masking behavior should be more intuitive and allow values like `7:00 am` as opposed to `07:00 am`

Note: there's a slightly less annoying behavior where if you press `1` in the first position for a time that is < `10`, the mask will expand and the cursor will move to the end.  If you go back to the proper position and continue typing your time, things will behave as expected -- a `:` will shrink the mask back to `1:__` while a number between `0` and `2` will input a double digit hour as expected.

### CODE REVIEW

File: src/components/tutor-input.cjsx [hunk 1]
```diff
@@ -212,15 +217,37 @@ TutorDateInput = React.createClass
...
-      m: '[0-9]'
-      P: '(A|P|a|p)'
-      p: '(M|m)'
     fromMomentFormat: 'HH:mm'
-    toMomentFormat: 'hh:mm a'
+    toMomentFormat: 'h:mm a'
+    formatCharacters:
+      i:
+        validate: (char) ->
+          /([0-2]|:)/.test(char)
```
Comment (philschatz) [inline src/components/tutor-input.cjsx]: Could these lines just be `validate: /([0-2]|:)/.test` ?

**Comment by philschatz** [comment]
Did the tests not need to be updated?

**Comment by pandafulmanda** [comment]
they do a little bit, but i think we need some other PRs to be merge first, and then #1026 with the multiple test updates to be merged

**Comment by philschatz** [comment]
yay it works!

**Pull request merged**